Sources/PushNotifications.swift

```diff
@@ -211,24 +211,26 @@ import Foundation
      - Precondition: `deviceToken` should not be nil.
      */
     /// - Tag: registerDeviceToken
-    @objc public func registerDeviceToken(_ deviceToken: Data, completion: @escaping () -> Void = {}) {
+    @objc public func registerDeviceToken(_ deviceToken: Data, completion: @escaping (Error?) -> Void = { _ in }) {
         guard
             let instanceId = Instance.getInstanceId(),
             let url = URL(string: "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances/\(instanceId)/devices/apns")
         else {
-            print("[Push Notifications] - Something went wrong. Please check your instance id: \(String(describing: Instance.getInstanceId()))")
+            completion(PushNotificationsError.error("[Push Notifications] - Something went wrong. Please check your instance id: \(String(describing: Instance.getInstanceId()))"))
             return
         }
 
         self.deviceToken = deviceToken
 
         if Device.idAlreadyPresent() {
             // TODO: Handle the token change.
+            completion(PushNotificationsError.error("Device is already present"))
```

Comment:
This isn't an error.

Reply:
Not necessarily an error, agreed. Maybe return a `state` enum instead? The state could have `.alreadyRegistered`, `registered`, ‘.notRegistered` - just a thought!

Reply:
This would mean adding additional argument besides the error in the completion handler.
I'll think about it.